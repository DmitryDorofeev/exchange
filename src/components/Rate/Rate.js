import React from 'react';
import {getCurrencySymbol} from '../../utils/currency';
import styles from './Rate.module.css';
import {Loader} from '../Loader/Loader';

export const Rate = ({from, to, rates, loading}) => {
  const rate = rates[from.currency] / rates[to.currency];
  const rateLine = `1${getCurrencySymbol(to.currency)} = ${rate.toFixed(2)}${getCurrencySymbol(from.currency)}`;

  return (
    <div className={styles.root}>
      {loading ? <Loader/> : rateLine}
    </div>
  )
};