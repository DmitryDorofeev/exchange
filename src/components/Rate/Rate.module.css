.root {
    display: flex;
    justify-content: center;
    align-items: center;
    border: 1px solid #ffffff;
    height: 42px;
    margin: 5px 0 15px 0;
    border-radius: 5px;
    color: #ffffff;
}
