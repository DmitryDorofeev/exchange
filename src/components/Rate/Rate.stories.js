import React from 'react';
import {Rate} from './Rate';
import {storiesOf} from '@storybook/react';

const rates = {
  USD: 1,
  GBP: 0.756179,
};
const from = {
  currency: 'GBP',
};
const to = {
  currency: 'USD',
};

storiesOf('Rate', module)
  .add('default', () => <Rate rates={rates} from={from} to={to}/>);