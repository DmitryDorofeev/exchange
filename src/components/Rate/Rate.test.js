import React from 'react';
import {shallow} from 'enzyme';
import {Rate} from './Rate';

describe('Rate component', () => {
  const rates = {
    USD: 1,
    GBP: 0.756179,
  };
  const from = {
    currency: 'GBP',
  };
  const to = {
    currency: 'USD',
  };
  const rate = shallow(<Rate rates={rates} from={from} to={to}/>);

  it('should match snapshot', () => {
    expect(rate).toMatchSnapshot();
  });
});