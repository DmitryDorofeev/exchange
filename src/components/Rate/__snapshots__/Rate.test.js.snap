// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rate
    from={
      Object {
        "currency": "GBP",
      }
    }
    rates={
      Object {
        "GBP": 0.756179,
        "USD": 1,
      }
    }
    to={
      Object {
        "currency": "USD",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "1$ = 0.76£",
      "className": "root",
    },
    "ref": null,
    "rendered": "1$ = 0.76£",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "1$ = 0.76£",
        "className": "root",
      },
      "ref": null,
      "rendered": "1$ = 0.76£",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
